# Disables pulseaudio support

+++ Makefile
@@ -79,10 +79,8 @@
 OBJS:=$(sort $(wildcard src/*.c *.c))
 OBJS:=$(OBJS:.c=.o)
 
-ifeq ($(OS),OpenBSD)
 OBJS:=$(filter-out src/pulse.o, $(OBJS))
 LIBS:=$(filter-out -lpulse, $(LIBS))
-endif
 
 src/%.o: src/%.c include/i3status.h
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
+++ src/print_volume.c
@@ -61,7 +61,8 @@
         free(instance);
     }
 
-#ifndef __OpenBSD__
+#define NO
+#ifndef NO
     /* Try PulseAudio first */
 
     /* If the device name has the format "pulse[:N]" where N is the
